ktor-utils/common/src/io/ktor/util/ByteChannels.kt

```diff
@@ -15,18 +15,20 @@ private const val CHUNK_BUFFER_SIZE = 4096L
  * Cancel of one channel in split(input or both outputs) cancels other channels.
  */
 @KtorExperimentalAPI
-public fun ByteReadChannel.split(coroutineScope: CoroutineScope): Pair<ByteReadChannel, ByteReadChannel> {
+public fun ByteReadChannel.split(): Pair<ByteReadChannel, ByteReadChannel> {
     val first = ByteChannel(autoFlush = true)
     val second = ByteChannel(autoFlush = true)
 
-    coroutineScope.launch {
+    GlobalScope.launch {
         try {
             while (!isClosedForRead) {
-                this@split.readRemaining(CHUNK_BUFFER_SIZE).use { chunk ->
-                    listOf(
-                        async { first.writePacket(chunk.copy()) },
-                        async { second.writePacket(chunk.copy()) }
-                    ).awaitAll()
+                supervisorScope {
```

Comment:
Does it mean that we changed the `split` behaviour? The fail of single channel led to fail the whole split

Reply:
reverted changes because the original issue is fixed in coroutines